FrontEnd/src/pages/Browse.tsx

```diff
@@ -13,47 +13,72 @@ import {
 import DownloadIcon from '@mui/icons-material/Download';
 import DeleteIcon from '@mui/icons-material/Delete';
 import EditIcon from '@mui/icons-material/Edit';
-import data from '../mock-table-data.json';
 import CircularProgress from '@mui/material/CircularProgress';
 import '../CSS/Browse.css';
+import SelectPlot from '../components/SelectPlot';
+import { plotProps } from '../components/SelectPlot';
 
-const Browse = () => {
-  let handleRemove = (i: any) => {
-    const newRows = [...rows];
-    const index = rows.findIndex((row) => row.file === i);
-    newRows.splice(index, 1);
-    setRows(newRows);
+interface FileWithMetadata {
+  fileName: {
+    metaData: string;
   };
+}
 
-  async function getData() {
-    return new Promise((resolve, reject) => {
-      setTimeout(() => {
-        resolve(data);
-        //reject('Failed to load');
-      }, 2000);
-    });
-  }
+const Browse = (props: plotProps) => {
+  // @TODO - implement remove and download files
+  // let handleRemove = (i: any) => {
+  //   const newRows = [...rows];
+  //   const index = rows.findIndex((row) => row.file === i);
+  //   newRows.splice(index, 1);
+  //   setRows(newRows);
+  // };
 
-  const [error, setError] = useState(null);
+  const [error, setError] = useState<Error>();
   const [isLoaded, setIsLoaded] = useState(false);
-  const [rows, setRows] = useState<any[]>([]);
+  const [rows, setRows] = useState<FileWithMetadata>();
+
+  const getListOfFiles = useCallback(async () => {
+    if (props.plot.plotName) {
+      const response = await fetch(
+        '/api/download?plot=' + props.plot.plotName,
+        {
+          method: 'Get',
+        }
+      );
+      const data = await response.json();
+      setRows(data);
+      setIsLoaded(true);
+    } else {
+      console.log('Plot is undefined');
+      setError(new Error('No plot'));
+    }
+  }, [props.plot.plotName]);
 
   useEffect(() => {
-    getData().then(
-      (data: any) => {
-        setIsLoaded(true);
-        setRows(data);
-      },
-      (error) => {
-        setIsLoaded(true);
-        setError(error);
-      }
-    );
-  }, []);
+    getListOfFiles();
+  }, [getListOfFiles]);
```

Comment:
The array you pass to `useEffect` is the list of dependencies that should re-trigger it. It doesn't make sense to pass a function here. 

What you want here is to pass an empty `[]` since you want this `useEffect` to run once, see: https://css-tricks.com/run-useeffect-only-once/